src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -636,42 +658,73 @@ private bool StartCore(ProcessStartInfo startInfo)
+                                        startInfo.UserName,
+                                        startInfo.Domain,
+                                        passwordPtr,
+                                        logonFlags,
+                                        null,            // we don't need this since all the info is in commandLine
+                                        commandLine,
+                                        creationFlags,
+                                        environmentPtr,
+                                        workingDirectory,
+                                        startupInfo,        // pointer to STARTUPINFO
+                                        processInfo         // pointer to PROCESS_INFORMATION
+                                    );
+                                if (!retVal)
+                                    errorCode = Marshal.GetLastWin32Error();
+                                if (processInfo.hProcess != IntPtr.Zero && processInfo.hProcess != (IntPtr)INVALID_HANDLE_VALUE)
+                                    procSH.InitialSetHandle(processInfo.hProcess);
+                                if (processInfo.hThread != IntPtr.Zero && processInfo.hThread != (IntPtr)INVALID_HANDLE_VALUE)
+                                    threadSH.InitialSetHandle(processInfo.hThread);
+                            }
                             if (!retVal)
-                                errorCode = Marshal.GetLastWin32Error();
-                            if (processInfo.hProcess != IntPtr.Zero && processInfo.hProcess != (IntPtr)INVALID_HANDLE_VALUE)
-                                procSH.InitialSetHandle(processInfo.hProcess);
-                            if (processInfo.hThread != IntPtr.Zero && processInfo.hThread != (IntPtr)INVALID_HANDLE_VALUE)
-                                threadSH.InitialSetHandle(processInfo.hThread);
+                            {
+                                if (errorCode == Interop.Errors.ERROR_BAD_EXE_FORMAT || errorCode == Interop.Errors.ERROR_EXE_MACHINE_TYPE_MISMATCH)
+                                {
+                                    throw new Win32Exception(errorCode, SR.InvalidApplication);
+                                }
+
+                                throw new Win32Exception(errorCode);
+                            }
                         }
-                        if (!retVal)
+                        finally
                         {
-                            if (errorCode == Interop.Errors.ERROR_BAD_EXE_FORMAT || errorCode == Interop.Errors.ERROR_EXE_MACHINE_TYPE_MISMATCH)
-                                throw new Win32Exception(errorCode, SR.InvalidApplication);
-
-                            throw new Win32Exception(errorCode);
+                            if (passwordPtr != IntPtr.Zero)
+                            {
+                                Marshal.ZeroFreeGlobalAllocUnicode(passwordPtr);
```

Comment:
Because yo'ure putting the plain text password into passwordPtr, it will get freed here even though it wasn't allocated with SecureStringToGlobalAllocUnicode.